docs/learn/learn-xcvm.md

```diff
@@ -8,28 +8,107 @@ slug: ../learn-xcvm
 ---
 
 At the core of XCM lies the Cross-Consensus Virtual Machine (XCVM). A “message” in XCM is an XCVM
-program. The XCVM is a state machine, state is kept track in Registers.
-
-It’s an ultra-high level non-Turing-complete computer whose instructions are designed to be roughly
-at the same level as transactions. Messages are one or more XCM instructions. The program executes
-until it either runs to the end or hits an error, at which point it finishes up and halts. An XCM
-executor following the XCVM specification is provided by Parity, and it can be extended or
-customized, or even ignored altogether and users can create their own construct that follows the
-XCVM spec.
-
-A _message_ in XCM is simply just a programme that runs on the `XCVM`: in other words, one or more
-XCM instructions. To learn more about the XCVM and the XCM Format, see the latest
-[blog post](https://medium.com/polkadot-network/xcm-the-cross-consensus-message-format-3b77b1373392)
+program, referred to as an **"XCM"** or **"XCMs"** for multiple messages. The XCVM is a
+register-based state machine. The state is tracked in domain-specific registers that hold
+information that is used and mutated along the execution of a particular message. Most of the XCM
+format comprises these registers and the instructions used to compose XCVM programs.
+
+The XCVM is an ultra-high-level non-Turing-complete computer whose instructions are designed to be
+roughly at the same level as transactions in terms of definition. Messages are one or more XCM
+instructions executed in order by the XCVM. An XCM is executed until it either runs to the end or
+hits an error, at which point it finishes up and halts.
+
+The primary implementation of the XCVM is the
```

Comment:
Let's not call it "primary". That would imply we never expect any other implementation to become more significant, which is an overtly political assertion that does not belong in a specification.

"The first implementation" would work just as well.